api/schema.capnp

```diff
@@ -60,38 +49,60 @@ struct JobInfo {
 
 
 interface Commit {
-  jobs  @0 () -> (jobs :List(JobInfo));
+  jobs         @0 () -> (jobs :List(JobInfo));
 
   jobOfVariant @1 (variant :Text) -> (job :OCurrent.Job);
 
-  refs @2 (hash :Text) -> (refs :List(Text));
+  refs         @2 (hash :Text) -> (refs :List(Text));
   # Get the set of branches and PRs with this commit at their head.
 
-  status @3 () -> (status :BuildStatus);
+  status       @3 () -> (status :BuildStatus);
+
+  message      @4 (hash :Text) -> (title :Text);
+}
+
+struct RefInfo {
+  ref          @0 :Text;
+  hash    @1 :Text;
+  status  @2 :BuildStatus;
+  started :union {
+    ts    @3 :Float64;
+    none  @4 :Void;
+  }
+  message @5 :Text;
+  # The state of the ref's head commit
 }
 
 interface Repo {
   refs         @0 () -> (refs :List(RefInfo));
   # Get the set of branches and PRs being monitored.
+  mainRef      @1 () -> (ref :RefInfo);
+  # Get the main branch, for UI display purposes
```

Comment:
I don't think this is correct (or at least in principle it isn't ideal). 

If I understand correctly (looking at https://capnproto.org/language.html), to maintain backward compatibility we should add new fields by appending them to the list of fields rather than inserting them.

Reply:
This is a question we need to discuss. Do we want to maintain backward compatibility? Considering we maintain both ends of the pipe, is it a problem if we break the compatibility for the UI capnp? I'm not opinionated at all on this, WDYT?

Reply:
Assuming we update `live-engine` and `live-www` at the same time, I don't believe it should be an issue.

Reply:
I was brought up in the school of distributed systems where making any assumptions about the clients of your API is fraught with danger :) 

You are both 💯 - We do have control over the client and the server here. I would encourage us to maintain backward compatibility on principle unless there is pain involved. In this case, does the position of the new field `mainRef` matter?

Reply:
In this case it matters yes :)